test/integration/consul-container/go.mod

```diff
@@ -1,8 +1,10 @@
 module github.com/hashicorp/consul/test/integration/consul-container
 
-go 1.22
+go 1.22.5
 
-toolchain go1.22.5
+toolchain go1.22.7
+
+replace github.com/hashicorp/consul => ../../..
```

Comment:
See note here: https://github.com/hashicorp/consul/pull/21657/files#r1806775582

`consul` brought in `tencentcloud` dependencies that conflicted w/ an incompatible new v1 module w/ a new path. Using a path replace of `consul` here should be fine since this isn't a public submodule, is only used for tests, and doesn't impact releases.